public/app/features/connections/pages/EditDataSourcePage.tsx

```diff
@@ -13,9 +18,30 @@ export function EditDataSourcePage() {
   const params = new URLSearchParams(location.search);
   const pageId = params.get('page');
   const { navId, pageNav } = useDataSourceSettingsNav();
-
+  const subTitle = pageNav.subTitle;
+  const dispatch = useDispatch();
+  const dataSource = useDataSource(uid);
+  const dsi = getDataSourceSrv()?.getInstanceSettings(uid);
+  const hasAlertingEnabled = Boolean(dsi?.meta?.alerting ?? false);
+  const isAlertManagerDatasource = dsi?.type === 'alertmanager';
+  const alertingSupported = hasAlertingEnabled || isAlertManagerDatasource;
+  const onNameChange = (name: string) => dispatch(setDataSourceName(name));
+  const onDefaultChange = (value: boolean) => dispatch(setIsDefault(value));
```

Comment:
Yeah, just like Torkel says I think it would make sense to keep these "implementation details" in the sub-components instead (in `<EditDataSourceSubtitle>` and in `<EditDataSourceTitle>`), because in that case the page component would remain "leaner" and the sub-components would be easier to reuse / move around.

This would also be helpful because these changes will also need to be "mirrored" to [connections/pages/EditDataSourcePage](https://github.com/grafana/grafana/blob/main/public/app/features/connections/pages/EditDataSourcePage.tsx), and like that we won't duplicate this logic.